x-pack/plugins/ingest_pipelines/server/routes/api/get.ts

```diff
@@ -16,34 +16,26 @@ const paramsSchema = schema.object({
 export const registerGetRoutes = ({
   router,
   license,
-  lib: { isEsError },
+  lib: { handleErrors },
 }: RouteDependencies): void => {
   // Get all pipelines
   router.get(
     { path: API_BASE_PATH, validate: false },
     license.guardApiRoute(async (ctx, req, res) => {
-      const { callAsCurrentUser } = ctx.core.elasticsearch.legacy.client;
+      const requestHandler = async () => {
+        const { callAsCurrentUser } = ctx.core.elasticsearch.legacy.client;
 
-      try {
         const pipelines = await callAsCurrentUser('ingest.getPipeline');
-
         return res.ok({ body: deserializePipelines(pipelines) });
-      } catch (error) {
-        if (isEsError(error)) {
-          // ES returns 404 when there are no pipelines
-          // Instead, we return an empty array and 200 status back to the client
-          if (error.status === 404) {
-            return res.ok({ body: [] });
-          }
+      };
 
-          return res.customError({
-            statusCode: error.statusCode,
-            body: error,
-          });
+      const customErrorHandler = (error: any) => {
+        if (error.statusCode === 404) {
+          return res.ok({ body: [] });
         }
+      };
 
-        return res.internalError({ body: error });
-      }
+      return handleErrors(res, requestHandler, customErrorHandler);
```

Comment:
I am not a big fan of this and would prefer the `compose` suggested by JL. But until then, here we could get away with 1 less argument by adding a catch on L28


```js
const pipelines = await callAsCurrentUser('ingest.getPipeline').catch(e => {
  if (error.statusCode === 404) {
    return [];
  }
  throw (e);
});
```